src/sas/qtgui/Calculators/GenericScatteringCalculator.py

```diff
@@ -1357,12 +1427,70 @@ def complete(self, input, update=None):
             out = numpy.hstack(out)
             self.data_to_plot = out
             logging.info('Gen computation completed.')
+
+        # if Beta(Q) Calculation has been requested, run calculation
+        if self.is_beta:
+            self.create_betaPlot()
+        
         self.cmdCompute.setText('Compute')
         self.cmdCompute.setToolTip("<html><head/><body><p>Compute the scattering pattern and display 1D or 2D plot depending on the settings.</p></body></html>")
         self.cmdCompute.clicked.disconnect()
         self.cmdCompute.clicked.connect(self.onCompute)
         self.cmdCompute.setEnabled(True)
         return
+    
+    def create_betaPlot(self):
+        """Carry out the compuation of beta Q using provided & calculated data
+        Returns a list of BetaQ values
+
+        """
+        
+        #Center Of Mass Calculation
+        CoM = self.centerOfMass()
+        self.data_betaQ = []
+
+        # Default values
+        xmax = self.qmax_x
+        xmin = self.qmax_x * _Q1D_MIN
+        qstep = self.npts_x
+
+        fQ = [0 for j in range(self.npts_x)]
+        currentQValue = []
+        for a in range(self.npts_x):
+            if(self.npts_x == 1):
+                currentQValue.append(xmin)
+            else: 
+                currentQValue.append(xmin + (xmax - xmin)/(self.npts_x-1)*a)
+        formFactor = self.data_to_plot
+
+        for a in range(self.npts_x):
+                       
+            for b in range(len(self.nuc_sld_data.pos_x)):
+                #atoms
+                atomName = str(self.nuc_sld_data.pix_symbol[b])
+                #Coherent Scattering Length of Atom
+                cohB = periodictable.elements.symbol(atomName).neutron.b_c
+
+                coordinates = [float(self.nuc_sld_data.pos_x[b]),float(self.nuc_sld_data.pos_y[b]),float(self.nuc_sld_data.pos_z[b])]
+                
+                relativeCoordinate = [0.0,0.0,0.0]
+                relativeCoordinate[0] = coordinates[0] - CoM[0]
+                relativeCoordinate[1] = coordinates[1] - CoM[1]
+                relativeCoordinate[2] = coordinates[2] - CoM[2]
+
+                magnitudeRelativeCoordinate = numpy.sqrt(relativeCoordinate[0]**2 + relativeCoordinate[1]**2 + relativeCoordinate[2]**2)
+    
+                fQ[a] +=  (cohB * (numpy.sin(currentQValue[a] * magnitudeRelativeCoordinate) / (currentQValue[a] * magnitudeRelativeCoordinate)))
+
+            #Beta Q Calculation
+            self.data_betaQ.append((fQ[a] **2)/(formFactor[a]))
+
+        #Scale Beta Q to 0-1
+        scalingFactor = self.data_betaQ[0]
+        for i in range (len(self.data_betaQ)):
+            self.data_betaQ[i] = (self.data_betaQ[i]/scalingFactor)
+
```

Comment:
Or just 
```
self.data_betaQ = [x/scalingFactor for x in self.data_betaQ]
```
without using the loop explicitly (list comprehension)

Alternatively, you could use numpy arrays for all your lists, making operations much easier.